pkg/experiment/metastore/index/index.go

```
@@ -0,0 +1,511 @@
+
+	metastorev1 "github.com/grafana/pyroscope/api/gen/proto/go/metastore/v1"
+)
+
+// NewIndex initializes a new metastore index.
+//
+// The index provides access to block metadata. The data is partitioned by time, shard and tenant. Partition identifiers
+// contain the time period referenced by partitions, e.g., "20240923T16.1h" refers to a partition for the 1-hour period
+// between 2024-09-23T16:00:00.000Z and 2024-09-23T16:59:59.999Z.
+//
+// Partitions are mostly transparent for the end user, though PartitionMeta is at times used externally. Partition
+// durations are configurable (at application level).
+//
+// The index requires a backing Store for loading data in memory. Data is loaded directly via LoadPartitions() or when
+// looking up blocks with FindBlock() or FindBlocksInRange().
+//
+// Data can be unloaded via StartCleanupLoop() which is meant to be called by the index owner and ran in a goroutine.
+func NewIndex(store Store, logger log.Logger, cfg *Config) *Index {
+	return &Index{
+		loadedPartitions: make(map[PartitionKey]*indexPartition),
+		allPartitions:    make([]*PartitionMeta, 0),
+		store:            store,
+		logger:           logger,
+		Config:           cfg,
+	}
+}
+
+// LoadPartitions reads all partitions from the backing store and loads the recent ones in memory.
+func (i *Index) LoadPartitions() {
+	i.partitionMu.Lock()
+	defer i.partitionMu.Unlock()
+
+	i.loadedPartitions = make(map[PartitionKey]*indexPartition)
+	i.allPartitions = make([]*PartitionMeta, 0)
+	for _, key := range i.store.ListPartitions() {
+		pMeta, err := i.store.ReadPartitionMeta(key)
+		if err != nil {
+			level.Error(i.logger).Log("msg", "error reading partition metadata", "key", key, "err", err)
+			continue
+		}
+		i.allPartitions = append(i.allPartitions, pMeta)
+		if pMeta.Ts.Add(i.Config.PartitionTTL).Before(time.Now()) {
+			// too old, will load on demand
+			continue
+		}
+		_, _ = i.getOrLoadPartition(pMeta)
+	}
+
+	i.sortPartitions()
+}
```

Is it safe to ignore errors here?